Add Travis CI and Poetry

lbernick opened the pull request.

obi1kenobi commented on .travis.yml:
```diff
@@ -0,0 +1,21 @@
+matrix:
+  include:
+  - name: "Lint and static analysis"
+    python: 3.6
```
Any particular reason not to run lint / static analysis in a newer Python version?

lbernick replied:
nope!

obi1kenobi commented on .travis.yml:
```diff
@@ -0,0 +1,21 @@
+      - poetry run ./scripts/copyright_line_check.sh
+      - poetry run black --check --config pyproject.toml .
+      - poetry run flake8 --config setup.cfg . d
+      - poetry run isort --check --settings-path setup.cfg -rc .
```
Consider adding `pylint` and `bandit` as well, possibly in a later PR.

lbernick replied:
having some trouble getting pylint to work. I think I'll make an issue as a TODO to add pylint, bandit, and mypy.

obi1kenobi commented:
Are the file deletions intentional?

lbernick commented:
> Are the file deletions intentional?

Yes, I deleted setup.py because it is no longer needed with poetry, and deleted one of the licenses because I accidentally added duplicate licenses with different filenames.

lbernick commented:
One issue with poetry that I don't love: https://github.com/python-poetry/poetry/issues/2073
since we are importing and using `__version__` in `model_metadata_v3`

obi1kenobi commented:
I think I'm missing a step here — I'm not sure how `__version__` relates to that issue.

lbernick commented:
> I think I'm missing a step here — I'm not sure how `__version__` relates to that issue.

Sorry, should have elaborated. The package version is specified once in pyproject.toml, but the package itself uses the variable `__version__` to include in the model metadata what version of the library serialized the model. If we want to bump the package version, we need to update it in two places, or use one of the alternate suggestions on the github issue linked above.

obi1kenobi commented:
I'm still confused. That Github issue seems to talk about an issue where package extras are not persisted by `poetry` across lockfiles. Here we're talking about defining our own package's version in only a single place. I'm sorry =/ I still don't see the connection between extras on dependencies of our package, and our own package's version.

lbernick commented:
Oops I linked the wrong issue! https://github.com/python-poetry/poetry/issues/144

obi1kenobi commented on .pylintrc:
```diff
@@ -0,0 +1,643 @@
+    # parameter_documentation, # needs some cleanup and maybe configuration
+    # unused-argument,         # needs some cleanup and per-line suppression,
+                               # buggy / unclear how to suppress only a single function
+    # unsubscriptable-object,  # TODO (Louis): Re-enable once https://github.com/PyCQA/pylint/issues/2822 is resolved
```
Heh which Louis? :D Maybe don't copy this TODO

obi1kenobi commented on .travis.yml:
```diff
@@ -0,0 +1,23 @@
+  - name: "Unit Tests"
+    python: 3.7
+    script:
+      - poetry run python -m pytest .
```
Perhaps in a separate PR, I'd add tests with all python versions that this package supports. Presumably that's 3.8 as well, and possibly 3.6 too?

lbernick commented:
@obi1kenobi @mikeyshulman finally got builds to pass! Right now it is using black, isort, flake8, and pydocstyle for linting, and we can potentially add pylint, mypy, and bandit later on. I also have it running tests for python 3.6 and 3.7 (TF doesn't yet support 3.8 it looks like).
